src/SetInterface.php

```diff
@@ -2,82 +2,18 @@
 
 namespace Dhii\Collection;
 
+use Traversable;
+use Dhii\Data\Container\HasCapableInterface;
+
 /**
- * Something that can behave as a set with complete functionality.
+ * A list that can be checked for a key.
  *
  * @since [*next-version*]
  */
-interface SetInterface extends \Countable
+interface SetInterface extends
```

Comment:
Looks very conceptually incorrect to me.

Here, it's declared as a traversable object which can be queried for a particular key's existence. I can't see how a consumer would require calling `has()` _and_ traversing. It seems to me like consumers would either need one or the other, in which case they ought to depend on `Traversable` or `HasCapableInterface` instead.